lib/Sema/CSBindings.cpp

```diff
@@ -981,7 +981,47 @@ bool BindingSet::isViable(PotentialBinding &binding) {
 }
 
 bool BindingSet::favoredOverDisjunction(Constraint *disjunction) const {
-  if (isHole() || isDelayed())
+  if (isHole())
+    return false;
+
+  if (llvm::any_of(Bindings, [&](const PotentialBinding &binding) {
+        if (binding.Kind == AllowedBindingKind::Supertypes)
+          return false;
+
+        auto type = binding.BindingType;
+
+        if (CS.shouldAttemptFixes())
+          return false;
+
+        if (type->isAnyHashable() || type->isDouble() || type->isCGFloatType())
```

Comment:
Yeah, we are typically super conservative and bail out of these sorts of heuristics if an implicit conversion is possible because it can affect inference in a lot of cases (especially with literals). Interestingly, only the CGFloat and Double conditions actually cause any tests to break when removed, but I think it's safest to keep this code as-is to start

Reply:
> Yeah, we are typically super conservative and bail out of these sorts of heuristics if an implicit conversion is possible because it can affect binding inference in a lot of cases (especially with literals).

Ah I see, thanks :)  

> Interestingly, only the CGFloat and Double conditions actually cause any tests to break when removed, but I think it's safest to keep this code as-is to start

Makes sense, do you think is worth a note comment for future us/others?

Reply:
Whoops, I merged before I read this. Sure, I can add a comment in a follow up (unless somebody else beats me to it) 🙂